### ISSUE TITLE
Application giving an error as The app encountered an error processing your request. If you are an administrator of the app, view the full error in the logs. See aka.ms/appservice-logs for more information. Error type: <class 'openai.AuthenticationError'>

### ISSUE THREAD

**Issue by javedunfyd** [Azure-Samples/azure-search-openai-demo#1070]
We ran the script as per the shared in "https://github.com/Azure-Samples/azure-search-openai-demo" and used existing resources. 
While sending any query to application, its giving an error stating "The app encountered an error processing your request.
If you are an administrator of the app, view the full error in the logs. See aka.ms/appservice-logs for more information.
Error type: <class 'openai.AuthenticationError'>"

![image](https://github.com/Azure-Samples/azure-search-openai-demo/assets/137041025/ef72c6f0-d98b-4634-b589-fa9ca626e593)
![service-assist-application-error](https://github.com/Azure-Samples/azure-search-openai-demo/assets/137041025/77186a11-cafb-487d-bf23-2a9f9de4c363)
![service-assist-application-error](https://github.com/Azure-Samples/azure-search-openai-demo/assets/137041025/b4c691fc-5b5e-43b0-be5c-84c4572c96b7)

**Comment by srbalakr** [comment]
Hi @javedunfyd
Could you confirm ?

1. You are on latest main
2. If an index has been created and contains data
    1. Remove any .md5 in /data/* .
    2. Run ./scripts/prepdocs.py from root directory.

**Comment by javedunfyd** [comment]
In data floder, there are lot of .md5 files. Shall I remove it?

**Comment by javedunfyd** [comment]
Run ./scripts/prepdocs.py from root directory. This also I have done but still there is no difference in error.

**Comment by srbalakr** [comment]
> In data floder, there are lot of .md5 files. Shall I remove it?

Yeah remove them and try running prepdocs.py again. md5 would prevent existing files to be picked up for new service and index.

**Comment by javedunfyd** [comment]
![image](https://github.com/Azure-Samples/azure-search-openai-demo/assets/137041025/9f90413d-a2a3-479e-9f02-a2802d44ac2c)

Here's giving an error.

**Comment by pamelafox** [comment]
The 404 error should be fine to ignore, that's only necessary if you have App Service authentication enabled, but I presume you don't (it's typically off).

In terms of openai.AuthenticationError, I suggest you check Application Insights for more information. Open Azure Portal, open the resource group for this app, open Application Insights, and select "Failures" under Investigate. You should see AuthenticationError listed, and you can click it for a longer traceback.

Most likely, something is misconfigured about your OpenAI connection. You can also open App Service in the Azure Portal and check the environment variables in the configuration to see how they look.
